inference-engine/src/mkldnn_plugin/utils/blob_dump.cpp

```diff
@@ -73,177 +76,131 @@ static TensorDesc parse_header(IEB_HEADER &header) {
         header.ver[1] != 1)
         IE_THROW() << "Dumper cannot parse file. Unsupported IEB format version.";
 
-    Precision prc = Precision(static_cast<Precision::ePrecision>(header.precision));
+    const auto prc = MKLDNNExtensionUtils::IEPrecisionToDataType(Precision(static_cast<Precision::ePrecision>(header.precision)));
     SizeVector dims(header.ndims);
     for (int i = 0; i < header.ndims; i++)
         dims[i] = header.dims[i];
 
-    return TensorDesc {prc, dims, TensorDesc::getLayoutByDims(dims) };
+    return MKLDNNMemoryDesc{MKLDNNDims(dims), prc, MKLDNNMemory::GetPlainFormatByRank(dims.size()) };
 }
 
-
-bool is_plain(const Blob::Ptr &blob) {
-    bool res = true;
-
-    auto orig_strides = blob->getTensorDesc().getBlockingDesc().getStrides();
-    auto orig_order = blob->getTensorDesc().getBlockingDesc().getOrder();
-    auto dims = blob->getTensorDesc().getDims();
-
-    for (int stride = 1, i = dims.size() - 1; i >= 0; --i) {
-        if (stride != orig_strides[i] || i != orig_order[i]) res = false;
-        stride *= dims[i];
-    }
-
-    return res;
+size_t product(const mkldnn::memory::dims &vector) {
+    return std::accumulate(vector.begin(), vector.end(), (size_t)1, std::multiplies<size_t>());
 }
 
-static Blob::Ptr prepare_plain_data(Blob::Ptr blob) {
-    // check if it already plain
-    if (is_plain(blob)) return blob;
+static void prepare_plain_data(const MKLDNNMemoryPtr &memory, std::vector<uint8_t> &data) {
+    const auto mdesc = memory->GetDescWithType<MKLDNNMemoryDesc>();
+    size_t data_size = product(memory->GetDims());
+    const auto &desc = memory->GetDesc();
```

Comment:
It looks like `desc` and `mdesc` are used interchangeably.

Reply:
mkldnn memory desc removed